neuralmonkey/learning_utils.py

```diff
@@ -277,13 +282,13 @@ def run_on_dataset(tf_manager: TensorFlowManager,
                                      compute_losses=contains_targets,
                                      batch_size=batch_size)
 
-    result_data_raw = {runner.output_series: result.outputs
-                       for runner, result in zip(runners, all_results)}
+    result_data = {runner.output_series: result.outputs
+                   for runner, result in zip(runners, all_results)}
 
     if postprocess is not None:
-        result_data = postprocess(dataset, result_data_raw)
-    else:
-        result_data = result_data_raw
+        for series_name, postprocessor in postprocess:
+            postprocessed = postprocessor(dataset, result_data)
+            result_data[series_name] = postprocessed
```

Comment:
Vida, to by se mohlo stát, kdyby `series_name` stejně pojmenovaná jako něco, co už leze z runneru. To by se asi mělo na začátku training_loop pohlídat.